src/org/rascalmpl/semantics/dynamic/PathTail.java

```diff
@@ -34,9 +42,23 @@ public Mid(ISourceLocation __param1, IConstructor tree, MidPathChars __param2, E
 		public Result<IValue> interpret(IEvaluator<Result<IValue>> __eval) {
 			Result<IValue> mid = this.getMid().interpret(__eval);
 			Result<IValue> expr = this.getExpression().interpret(__eval);
-			Result<IValue> tail = this.getTail().interpret(__eval);
 
-			return mid.add(expr).add(tail);
+			IString path = (IString) ResultFactory.makeResult(TF.stringType(), VF.string(""), __eval)
+				.add(expr).getValue();
+			
+			try {
```

Comment:
this part should also get some comments explaining what is happening.

Reply:
done